src/mapwindow.jl

```diff
@@ -273,13 +365,100 @@ end
 # This is slightly faster than a circular buffer
 @inline cyclecache(b, x) = b[1], (Base.tail(b)..., x)
 
+
 replace_function(f) = f
-replace_function(::typeof(median!)) = function(v)
-    inds = indices(v,1)
-    Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+
+replace_function(::typeof(median!)) = function(v,m_histogram,mode,window)
+   # Handle the boundary points with mode -1
+    if(mode==-1)
+        inds = indices(v,1)
+        return Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+    else
+        
+        window_size=size(v,1)
+        dims = map(x->x.stop-x.start+1,window)
+        width=window[1].stop-window[1].start+1
+        inds = indices(v,1)
+        if mode == 0
+        # Update the histogram according to new entries
+            for i = first(inds):last(inds)
+                id= trunc(Int64,(v[i]*255))+1
+                m_histogram[id]+=1
+            end
+            counter=0
+
+            for i =1:256
+                counter+=m_histogram[i]
+            end
+        # Compute the median
+            tempsum = 0
+            m_index=-1
+            for i = 1:256
+                tempsum+= m_histogram[i]
+                if tempsum>=trunc(Int64,window_size/2)+1
+                    m_index=i-1
+                    break
+                end
+            end
+        # Clear the histogram from previous value
+            for i = first(inds):width:dims[1]*dims[2]
+                for j= i: dims[1]*dims[2]:last(inds)
+                    println(j)    
+                    id= trunc(Int64,(v[j]*255))+1
+                    m_histogram[id]-=1
+                    if(m_histogram[id]<0)
+                        println("stop")
+                    end
+                end
+            end
+            return convert(Float64,m_index)/255
+
+        elseif mode == 1
+        # Update the histogram according to new entries
+            for i =  width:width:dims[1]*dims[2]
```

Comment:
No, this isn't 2D. The iterator i iterates through all the new values added in all dimensions .  
''
for i = first(inds):width:dims[1]*dims[2] [This iterates over a 2D slice]  
                for j= i: dims[1]*dims[2]:last(inds)  [This iterates over all such 2D slices in nD]
''